src/VirtoCommerce.CustomerSegmentsModule.Core/Models/CustomerSegmentTree.cs

```diff
@@ -1,12 +1,38 @@
+using System.Collections.Generic;
 using VirtoCommerce.CoreModule.Core.Conditions;
+using VirtoCommerce.CustomerModule.Core.Model.Search;
+using VirtoCommerce.Platform.Core.Common;
 
 namespace VirtoCommerce.CustomerSegmentsModule.Core.Models
 {
-    public class CustomerSegmentTree : BlockConditionAndOr
+    public class CustomerSegmentTree : BlockConditionAndOr, ICanBuildSearchCriteria
     {
         public CustomerSegmentTree()
         {
             All = true;
         }
+
+        public virtual MembersSearchCriteria BuildSearchCriteria(IMemberSearchCriteriaBuilder builder)
+        {
+            if (!Children.IsNullOrEmpty())
+            {
+                BuildSearchCriteria(Children, builder);
```

Comment:
Why do we pass base property to the protection method which already has access to the base property? Looks like a redundant parameter.